packages/nocodb/src/controllers/api-docs/api-docs.controller.ts

```diff
@@ -33,11 +33,15 @@ export class ApiDocsController {
 
   @Get('/api/v1/db/meta/projects/:projectId/swagger')
   swaggerHtml(@Param('projectId') projectId: string, @Response() res) {
-    res.send(getSwaggerHtml({ ncSiteUrl: process.env.NC_PUBLIC_URL || '' }));
+    const ncPublicUrl = process.env.NC_PUBLIC_URL || '';
+    const swaggerHtml = getSwaggerHtml({ ncSiteUrl : ncPublicUrl })
+    res.send(swaggerHtml);
```

Comment:
How does it fix the issue with such changes? You just break that line into 3 lines. Besides, your first statement is not correct, `process.env.NC_PUBLIC_URL || ''` is already used. Also there is nothing related to `NC_BACKEND_URL`.

Reply:
> 

With this modification, the Swagger and ReDoc documentation links will use NC_PUBLIC_URL as the base URL for generating the correct links. and variable NC_PUBLIC_URL. If the variable is not defined or empty, it will default to an empty string.